Fix runtime pointer direct cast

hcindyl opened the pull request:
Add `intptr_t` cast in between `uint64_t` and the `void*` or flatbuffer pointer types.


To void the `int-to-pointer-cast` and `pointer-to int-cast` warnings when build with 32-bit system

```
iree/runtime/src/iree/hal/allocator_heap.c:237:13: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  237 |       ptr = (void*)external_buffer->handle.device_allocation.ptr;
      |             ^

```
```
iree/runtime/src/iree/vm/bytecode/module.c:528:7: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  528 |       (iree_vm_FunctionSourceMapDef_table_t)data[0];
      |       ^

iree/vm/bytecode/module.c:590:34: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  590 |   out_source_location->data[0] = (uint64_t)source_map_def;
      |                                  ^

```

benvanik reviewed (changes requested):
these need to be uint64_t and not variable-length - the warnings should be fixed when casting and not by changing the types of the API structures.

try instead changing where the warnings were emitted to first cast to `intptr_t` before casting to `void*` or the flatbuffers type

hcindyl committed:
Fix runtime pointer direct cast

Add `intptr_t` cast in between `uint64_t` and the `void*` or flatbuffer
pointer types.

Change-Id: I714af1cbc92fed011e507256fb23b9de48b2d8b7
